tests/test_rtdc_feat_anc_plugin.py

```diff
@@ -905,6 +958,160 @@ def test_pf_wrong_length_2():
+    # Mention feature shape in the info
+    info["feature shapes"] = [(80, 250)]
+    PlugInFeature("mask_feat", info)
+    with dclab.new_dataset(h5path) as ds:
+        mask_feat = np.logical_not(ds["mask"])
+    with RTDCWriter(h5path) as hw:
+        hw.store_feature("mask_feat", mask_feat)
+    with h5py.File(h5path, mode="r") as h5:
+        events = h5["events"]
+        assert "mask_feat" in events.keys()
+        feat = h5["events"]['mask_feat']
+        fshape = feat.shape
+        assert feat.ndim == 3
+        assert fshape[1] > 3 and fshape[2] > 3
+        assert b'CLASS' in feat.attrs.keys()
+        assert b'IMAGE_SUBCLASS' in feat.attrs.keys()
+        assert b'IMAGE' in feat.attrs.get('CLASS')
+
+
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_pf_store_mask_plugin_data_without_shape_issue_171():
+    """Test storage of mask plugin feature attributes without shape info"""
+    h5path = retrieve_data("fmt-hdf5_fl_2018.zip")
+    info = example_plugin_info_mask_feature()
+    PlugInFeature("mask_feat", info)
+    with dclab.new_dataset(h5path) as ds:
+        mask_feat = np.logical_not(ds["mask"])
+    with RTDCWriter(h5path) as hw:
+        hw.store_feature("mask_feat", mask_feat, shape=(80, 250))
+    with h5py.File(h5path, mode="r") as h5:
+        events = h5["events"]
+        assert "mask_feat" in events.keys()
+        feat = h5["events"]['mask_feat']
+        fshape = feat.shape
+        assert feat.ndim == 3
+        assert fshape[1] > 3 and fshape[2] > 3
+        assert b'CLASS' in feat.attrs.keys()
+        assert b'IMAGE_SUBCLASS' in feat.attrs.keys()
+        assert b'IMAGE' in feat.attrs.get('CLASS')
+
+
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_pf_store_ragged_plugin_data_issue_171():
+    """Test storage of ragged plugin feature"""
+    h5path = retrieve_data("fmt-hdf5_fl_2018.zip")
+    info = example_plugin_info_ragged_data_feature()
+    with dclab.new_dataset(h5path) as ds:
+        cnts = ds["contour"]  # shape --> [None, nan, 2]
+        # Add dummy contour coordinates to the actual contours
+        ragged_feat = np.array([np.vstack((c, np.ones((3, 2)))) for c in cnts],
+                               dtype=object)
+    PlugInFeature("ragged_feat", info)
+    with RTDCWriter(h5path) as hw:
+        hw.store_feature("ragged_feat", ragged_feat)
+    with h5py.File(h5path, mode="r") as h5:
+        events = h5["events"]
+        assert "ragged_feat" in events.keys()
+        assert np.allclose(events["ragged_feat"]["1"], ragged_feat[1])
+        assert np.allclose(events["ragged_feat"]["6"], ragged_feat[6])
+        assert events["ragged_feat"]["1"].shape == (57, 2)
+        assert events["ragged_feat"]["6"].shape == (52, 2)
+
+
+@pytest.mark.filterwarnings(
+    "ignore::dclab.rtdc_dataset.config.WrongConfigurationTypeWarning")
+def test_pf_store_scalar_plugin_data_issue_171():
```

Comment:
Best to put the scalar functions above the non-scalar, makes more sense in my brain at least